sample/sample.go

```diff
@@ -17,6 +21,68 @@ var (
 	badLengthMismatch = "sample: slice length mismatch"
 )
 
+var (
+	_ Sampler = LatinHypercuber{}
+	_ Sampler = MetropolisHastingser{}
+	_ Sampler = Rejectioner{}
+	_ Sampler = IIDer{}
+
+	_ WeightedSampler = WeightedSample{}
+	_ WeightedSampler = Importancer{}
+)
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// Sampler generates a batch of samples according to the rule specified by the
+// implementing type. The number of samples generated is equal to the length of
+// the input slice, and the samples are stored in-place into the input.
+type Sampler interface {
+	Sample(samples []float64)
+}
+
+// WeightedSampler generates a batch of samples and their relative weights
+// according to the rule specified by the implementing type. The number of samples
+// generated is equal to the length of the input slice, and the samples and weights
+// are stored in-place into the inputs. The length of weights must equal the
+// length of samples, otherwise the WeightedSampler will panic.
+type WeightedSampler interface {
+	SampleWeighted(samples, weights []float64)
+}
+
+// WeightedSample wraps a sampler type to create a WeightedSampler where all
+// weights are equal.
+type WeightedSample struct {
+	Sampler
+}
+
+func (w WeightedSample) SampleWeighted(samples, weights []float64) {
+	if len(samples) != len(weights) {
```

Comment:
I think it can't be in this case. The point of weights is that some sampling methods MUST have a weight in order to be statistically valid. Importance sampling, for example, must have each sample corrected by its relative probability. The assumption is that if weights is not stored, the method cannot be being used properly